src/db/statement.cr

```diff
@@ -0,0 +1,112 @@
+module DB
+  # Represents a prepared query in a `Connection`.
+  # It should be created by `QueryMethods`.
+  #
+  # ### Note to implementors
+  #
+  # 1. Subclass `Statements`
+  # 2. `Statements` are created from a custom driver `Connection#prepare` method.
+  # 3. `#perform_query` executes a query that is expected to return a `ResultSet`
+  # 4. `#perform_exec` executes a query that is expected to return an `ExecResult`
+  # 6. `#do_close` is called to release the statement resources.
+  abstract class Statement
+    include Disposable
+
+    # :nodoc:
+    getter connection
+
+    def initialize(@connection)
+    end
+
+    protected def do_close
```

Comment:
I think is not safe to do that. If in the future Statement#do_close needs to do something implementors needs to add the call. If they add them now everything will work tomorrow.

Making it empty now also unifies all Disposables for implementors.

Right a `super` that binds to an abstract method fails to compile, so there is not smooth transition possible from a base abstract class to decide later to add an implementation.